File: 99-Continue/163/RTL.py
from collections import deque

class Node:
    def __init__(self, data):
        self.data = data
        self.left = self.right = None


class BinaryTree:
    def __init__(self):
        self.root = None

    def buildTree(self, arr):
        if not arr:
            return None

        self.root = Node(int(arr[0]))
        queue = deque([self.root])

        i = 1

        while queue and i < len(arr):
            current = queue.popleft()

            if i < len(arr) and arr[i] != "null":
                current.left = Node(int(arr[i]))
                queue.append(current.left)
            i += 1

            if i < len(arr) and arr[i] != "null":
                current.right = Node(int(arr[i]))
                queue.append(current.right)
            i += 1

        return self.root

    def rootToLeaf(self, root, s, lst):
        if root is None:
            return

        s += str(root.data)

        if root.left:
            self.rootToLeaf(root.left, s, lst)

        if root.right:
            self.rootToLeaf(root.right, s, lst)

        if root.left is None and root.right is None:
            lst.append(int(s))


def main():
    input()  # n
    arr = input().split()

    tree = BinaryTree()
    tree.buildTree(arr)

    lst = []
    tree.rootToLeaf(tree.root, "", lst)

    print(sum(lst))


if __name__ == "__main__":
    main()

File: 99-Continue/163/RTL.java
import java.util.*;

public class RTL {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();
        String s = sc.nextLine();
        String[] arr = s.split(" ");
        List<Integer> lst = new ArrayList<>();
        BinaryTree tree = new BinaryTree();
        tree.buildTree(arr);
        tree.rootToLeafSum(tree.root, "",lst);
        int sum = 0;
        for(int i:lst){
            sum+=i;
        }
        System.out.println(sum);
        sc.close();
    }
}

class Node{
    Node left,right;
    int data;
    Node(int data){
        this.data = data;
        this.left = this.right = null;
    }
}

class BinaryTree{
    Node root;
    BinaryTree(){
        this.root = null;
    }
    
    Node buildTree(String[] arr){
        if(arr.length ==0){
            return root;
        }
        Queue<Node> queue = new LinkedList<>();
        root=new Node(Integer.parseInt(arr[0]));
        queue.offer(root);
        int i = 1;
        while(!queue.isEmpty()&&i<arr.length){
            Node current = queue.poll();
            if(i<arr.length&&!arr[i].equals("null")){
                current.left = new Node(Integer.parseInt(arr[i]));
                queue.offer(current.left);
            }
            i++;
            if(i<arr.length&&!arr[i].equals("null")){
                current.right = new Node(Integer.parseInt(arr[i]));
                queue.offer(current.right);
            }
            i++;
        }
        return root;
    }
    
    
    
    void rootToLeafSum(Node root,String s,List<Integer> lst){
        if(root==null) return;
        s+=root.data;
        if(root.left!=null){
            rootToLeafSum(root.left,s,lst);
        }
        if(root.right!=null){
            rootToLeafSum(root.right,s,lst);
        }
        
        if(root.left==null && root.right==null){
           lst.add(Integer.parseInt(s));
        }
        
    }
}